File: 1/2020.10.16/55.Jump Game/java/solution.java
class Solution {
    public boolean canJump(int[] nums) {
        if (nums.length < 2) {
            return true;
        }
        int maxPos = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (i <= maxPos) {
                maxPos = Math.max(maxPos, i + nums[i]);
                if (maxPos >= nums.length - 1) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: 1/2020.10.16/55.Jump Game/python/solution.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if len(nums) < 2:
            return True
        max_pos = nums[0]
        for i in range(1, len(nums)):
            if i <= max_pos:
                max_pos = max(max_pos, i + nums[i])
                if max_pos >= len(nums) - 1:
                    return True
        return False
